doc/building_picoquic.md

```diff
@@ -0,0 +1,102 @@
+# Building PicoQUIC
+
+See the 'Building Picoquic' section [README](../README.md) for building picoquic for the first time and installing the dependencies.
+
+## Dependencies
+
+Direct dependencies
+- [Picotls implementation of TLS 1.3](https://github.com/h2o/picotls)
+
+Fetching and installing this package can be done through an option in the CMake.
+
+```shell
+ cmake -DPICOQUIC_FETCH_PTLS=Y .
+```
+
+This can be set in the cmake file as well by setting the `PICOQUIC_FETCH_PTLS` cmake property to `ON`.
+
+Inherited dependencies
+- [OpenSSL through PicoTLS](https://github.com/openssl/openssl)
+
+TODO: Current issues and ongoing work
+
+There is a current issue trying to enable the use of `mbedtls` instead of `openssl`.
+
+## Targets
+
+Targets are anything that is defined as an executable in `CMakeLists.txt`.
+Available targets:
+
+- picoquicdemo
+- picolog_t
+- picoquic_ct
+- picohttp_ct
+- pico_baton
+- picoquic_sample
+- thread_test
+
+All of these targets are built when the `make .` or `cmake --build .` commands are used to build the project. After which the test program `picoquic_ct` can be used to verify the port.
+
+`picoquicdemo` ([found in picoquicfirst/picoquicdemo.c](../picoquicfirst/picoquicdemo.c)) and `picoquic_sample` ([sample documentation](../sample/README.md)) are a good starting point for developing an application on top of this QUIC implementation.
+
+
+## (Re)Building a Single Target
+
+Targets defined in the `CMakeLists.txt`, listed above for convenience, can be build individually using cmake's `--target` option. The command `cmake --build . --target <target_name>` can be used to build a specific target and all it's dependcies. For example `cmake --build . --target picoquic_sample` will build the sample file transfer application in [sample/](../sample/README.md) as defined in the `CMakeLists.txt`. All the required components will also be built. This works if the command is issued in the top-level directory of the repository. This is useful for recompiling test programs and programs like `picoquic_sample` and `picoquicdemo` if they have been modified. Therefore reducing build times when making small changes.
+
+This also works if `picoquic` has not been built as a whole before.
+
+## Picoquic options
+
+- building and fetching PTLS `-DPICOQUIC_FETCH_PTLS=Y`
+- building with exported debug information `-DCMAKE_BUILD_TYPE=Debug`
+
+These two options can be set when configuring cmake.
+
+- adress sanitation `ASAN`
+- undefined behaviour sanitation `UBSAN`
+
+`ASAN` and `UBSAN` need to be set in the `CMakeLists.txt` using the `ENABLE_ASAN` and `ENABLE_UBSAN` options.
+
```

Comment:
There is no need to edit the CMake file, and I would rather not encourage people to do that. The correct text should be something like:

`ASAN` and `UBSAN` are C and C++ compiler options, and can be instantiated by setting compiler flags when launching `cmake`, as in:
```shell
cmake "-DCMAKE_C_FLAGS=-fsanitize=address,undefined -DCMAKE_CXX_FLAGS=-fsanitize=address,undefined"
```